### PR TITLE
Support hdfs for lake table

### PR DESCRIPTION
**Pull request by tracymacding**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

File: be/src/service/staros_worker.cpp [hunk 1]
```diff
@@ -16,20 +16,14 @@ namespace starrocks {
 
 namespace fslib = staros::starlet::fslib;
 
-std::ostream& operator<<(std::ostream& os, const staros::starlet::ShardInfo& shard) {
-    return os << "Shard{.id=" << shard.id << " .uri=" << shard.obj_store_info.s3_obj_store.uri << "}";
-}
-
 absl::Status StarOSWorker::add_shard(const ShardInfo& shard) {
     std::unique_lock l(_mtx);
-    LOG(INFO) << "Adding " << shard;
```
Comment (kevincai) [inline be/src/service/staros_worker.cpp:25]: this line is helping troubleshooting a lot, may still need to keep it

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // This file is made available under Elastic License 2.0.
-// This file is based on code available under the Apache license here:
+// This file is based on code init() under the Apache license here:
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:2]: seems unnecessary change?

**Review by kevincai — COMMENTED** [review]
missing POM starmanager/starproto version change?

**Comment by starrocks-xupeng** [comment]
how can you pass compile if you don't add 0.1.9?

File: be/src/service/staros_worker.cpp (hunk 1 above)
Comment (tracymacding) [reply be/src/service/staros_worker.cpp:25]: starlet has already log this infomation

File: fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java (hunk 2 above)
Comment (tracymacding) [reply fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java:2]: Maybe i made a mistake

Comment (tracymacding) [reply fe/fe-core/src/main/java/com/starrocks/server/GlobalStateMgr.java]: done

**Review by kevincai — DISMISSED** [review]

**Review by starrocks-xupeng — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by tracymacding** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 49 (73.47%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/backup/LakeRestoreJob.java|0|2|00.00%|[294, 297]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|2|8|25.00%|[102, 103, 104, 105, 106, 108]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|8|12|66.67%|[75, 157, 158, 162]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|3|4|75.00%|[84]|
|:large_blue_circle:|com/starrocks/lake/StorageCacheInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/StorageInfo.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/staros/StarMgrServer.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/lake/StorageInfo.java [hunk 3]
```diff
@@ -14,12 +14,17 @@
 // Currently, storage group is table level.
 public class StorageInfo implements GsonPreProcessable, GsonPostProcessable {
     // "shardStorageInfoBytes" is used for serialization of "shardStorageInfo".
```
[the remark this answers is not in the document]
Comment (tracymacding) [reply fe/fe-core/src/main/java/com/starrocks/lake/StorageInfo.java]: done

**Review by wyb — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by wyb** [comment]
run starrocks_admit_test

**Comment by wyb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 36 / 49 (73.47%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/backup/LakeRestoreJob.java|0|2|00.00%|[294, 297]|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|2|8|25.00%|[102, 103, 104, 105, 106, 108]|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|8|12|66.67%|[75, 157, 158, 162]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|3|4|75.00%|[84]|
|:large_blue_circle:|com/starrocks/lake/StorageCacheInfo.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/Config.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/lake/StorageInfo.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/staros/StarMgrServer.java|1|1|100.00%|[]|
